### ISSUE TITLE
Kubernetes unable to run some containers with `ImageInspectError`

### ISSUE THREAD

**Issue by campbel** [rancher-sandbox/rancher-desktop#7160]
### Actual Behavior

Some container images, including Rancher Desktop Kubernetes defaults, are unable to run because of an `ImageInspectError`.

### Steps to Reproduce

1. Download and install Rancher Desktop 1.14.2
2. Enable Kubernetes (wait for it to boot)
3. Run `kubectl get pods -A`

This is sufficient to reproduce on my system consistently.

### Result

```bash
kubectl get pods -A
NAMESPACE     NAME                                      READY   STATUS              RESTARTS   AGE
kube-system   coredns-6799fbcd5-bmswr                   0/1     ImageInspectError   0          63s
kube-system   local-path-provisioner-84db5d44d9-4xmlp   0/1     ImageInspectError   0          63s
kube-system   metrics-server-67c658944b-sxwst           0/1     ImageInspectError   0          63s
```

```bash
kubectl -n kube-system describe pod coredns-6799fbcd5-bmswr
Name:                 coredns-6799fbcd5-bmswr
Namespace:            kube-system
Priority:             2000000000
Priority Class Name:  system-cluster-critical
Service Account:      coredns
Node:                 lima-rancher-desktop/192.168.5.15
Start Time:           Mon, 08 Jul 2024 19:37:54 -0600
Labels:               k8s-app=kube-dns
                      pod-template-hash=6799fbcd5
Annotations:          <none>
Status:               Pending
IP:                   10.42.0.2
IPs:
  IP:           10.42.0.2
Controlled By:  ReplicaSet/coredns-6799fbcd5
Containers:
  coredns:
    Container ID:  
    Image:         rancher/mirrored-coredns-coredns:1.10.1
    Image ID:      
    Ports:         53/UDP, 53/TCP, 9153/TCP
    Host Ports:    0/UDP, 0/TCP, 0/TCP
    Args:
      -conf
      /etc/coredns/Corefile
    State:          Waiting
      Reason:       ImageInspectError
    Ready:          False
    Restart Count:  0
    Limits:
      memory:  170Mi
    Requests:
      cpu:        100m
      memory:     70Mi
    Liveness:     http-get http://:8080/health delay=60s timeout=1s period=10s #success=1 #failure=3
    Readiness:    http-get http://:8181/ready delay=0s timeout=1s period=2s #success=1 #failure=3
    Environment:  <none>
    Mounts:
      /etc/coredns from config-volume (ro)
      /etc/coredns/custom from custom-config-volume (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-lkq7h (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      coredns
    Optional:  false
  custom-config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      coredns-custom
    Optional:  true
  kube-api-access-lkq7h:
    Type:                     Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:   3607
    ConfigMapName:            kube-root-ca.crt
    ConfigMapOptional:        <nil>
    DownwardAPI:              true
QoS Class:                    Burstable
Node-Selectors:               kubernetes.io/os=linux
Tolerations:                  CriticalAddonsOnly op=Exists
                              node-role.kubernetes.io/control-plane:NoSchedule op=Exists
                              node-role.kubernetes.io/master:NoSchedule op=Exists
                              node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                              node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Topology Spread Constraints:  kubernetes.io/hostname:DoNotSchedule when max skew 1 is exceeded for selector k8s-app=kube-dns
Events:
  Type     Reason         Age                   From               Message
  ----     ------         ----                  ----               -------
  Normal   Scheduled      4m3s                  default-scheduler  Successfully assigned kube-system/coredns-6799fbcd5-bmswr to lima-rancher-desktop
  Warning  Failed         116s (x12 over 4m3s)  kubelet            Error: ImageInspectError
  Warning  InspectFailed  102s (x13 over 4m3s)  kubelet            Failed to inspect image "rancher/mirrored-coredns-coredns:1.10.1": Id or size of image "rancher/mirrored-coredns-coredns:1.10.1" is not set

### Expected Behavior

Containers are able to boot and run as expected.

### Additional Information

_No response_

### Rancher Desktop Version

1.14.1

### Rancher Desktop K8s Version

1.28.4

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

Sonoma 14.5

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

_No response_

**Comment by campbel** [comment]
As a workaround, I did a factory reset and re-setup Rancher Desktop using latest Kubernetes (1.29.6 - stable) and it does seem to boot properly now. Could this be an issue with the version of Kubernetes we were on or cached images?

**Comment by jandubois** [comment]
> Could this be an issue with the version of Kubernetes we were on

Yes. I think the root cause is that we are using the version of `cri-dockerd` bundled with `k3s`, and the older versions of that shim are not compatible with Docker API 1.44, which is used by the newer `docker` CLI bundled with the latest Rancher Desktop releases.

Essentially the same as https://github.com/k3s-io/k3s/issues/9279

The fix for Rancher Desktop should be to use an updated bundled version of `cri-dockerd` and ignore the version bundled with `k3s`.

As a user switching to a newer version of Kubernetes (or using `containerd` instead of `moby`) are the currently available workarounds.

**Comment by jandubois** [comment]
This will be fixed in `1.15.0`.

For the record, these are the `k3s` versions that don't work with `moby` in `1.14.x`:

```
       '1.26.8 - 1.26.13',
       '1.27.5 - 1.27.10',
       '1.28.0 - 1.28.6',
       '1.29.0 - 1.29.1',
```
